src/default-client/index.ts

```diff
@@ -0,0 +1,840 @@
+      ast,
+      fieldTypeMap,
+      result.data,
+      createCacheMetadata(result.cacheMetadata),
+      { cacheResolve: deferred.resolve, tag: opts.tag },
+    );
+
+    if (opts.awaitDataCached) await deferred.promise;
+
+    return this._resolve({
+      cacheMetadata: resolveResult.cacheMetadata,
+      data: resolveResult.data,
+      operation: "subscription",
+    });
+  }
+
+  private _setPendingRequest(queryHash: string, { reject, resolve }: PendingRequestActions): void {
+    let pending = this._cache.requests.pending.get(queryHash);
+    if (!pending) pending = [];
+    pending.push({ reject, resolve });
+    this._cache.requests.pending.set(queryHash, pending);
+  }
+
+  private async _subscription(
+    subscription: string,
+    ast: DocumentNode,
+    opts: RequestOptions,
+    context: RequestContext,
+  ): Promise<ResolveResult | AsyncIterator<Event | undefined>> {
+    const hash = hashRequest(subscription);
+
+    try {
+      return this._subscriptionService.send(
+        subscription,
+        hash,
+        async (result) => this._resolveSubscriber(ast, context.fieldTypeMap, result, opts),
+      );
+    } catch (error) {
+      return this._resolve({ error, operation: "subscription" });
+    }
+  }
+
+  private async _updateQuery(
+    query: string,
+    opts: RequestOptions,
+    context: RequestContext,
+  ): Promise<{ ast: DocumentNode, query: string }> {
+    const _this = this;
+    const typeInfo = new TypeInfo(this._schema);
+    let fragmentDefinitions: FragmentDefinitionNodeMap | undefined;
+
+    const ast = visit(parse(query), {
+      enter(
+        node: ASTNode,
+        key: string | number,
+        parent: any,
+        path: Array<string | number>,
+        ancestors: any[],
+      ): ValueNode | undefined {
+        typeInfo.enter(node);
+        const kind = getKind(node);
+
+        if (kind === "Document") {
+          const documentNode = node as DocumentNode;
+          if (!opts.fragments || !hasFragmentDefinitions(documentNode)) return undefined;
+          fragmentDefinitions = getFragmentDefinitions(documentNode);
+          deleteFragmentDefinitions(documentNode);
+          return undefined;
+        }
+
+        if (kind === "Field" || kind === "InlineFragment") {
+          const fieldOrInlineFragmentNode = node as FieldNode | InlineFragmentNode;
+          const type = _this._getFieldOrInlineFragmentType(kind, fieldOrInlineFragmentNode, typeInfo);
+          if (!type) return undefined;
+
+          if (kind === "Field") {
+            const fieldNode = node as FieldNode;
+
+            if (fragmentDefinitions && hasFragmentSpreads(fieldNode)) {
+              setFragmentDefinitions(fragmentDefinitions, fieldNode);
+            }
+          }
+
+          if (!(type instanceof GraphQLObjectType) && !(type instanceof GraphQLInterfaceType)) return undefined;
+          const objectOrInterfaceType = type as GraphQLObjectType | GraphQLInterfaceType;

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS4325)